### ISSUE TITLE
Crash when opening 'Attach' window

### ISSUE THREAD

**Issue by HydrophonePia** [x64dbg/x64dbg#3075]
### Operating System

W10 22H2 19045.2364

### x64dbg Version

Apr 27 2023

### Describe the issue

Love the debugger, so thought I'd report this bug(missing nullptr check).
Attach(Alt+A keybind) in x64dbg, under some circumstances, crashes due to bugs in src/dbg/debugger.cpp
the problem is, when argv does not contain image path(for example, if argv[0] is some invalid shit), function dbglistprocesses in debugger.cpp assumes that exeName contains a dot(which it doesn't have to), and if there is no dot - the debugger simply crashes(nullptr dereference)
relevant code:
[https://github.com/x64dbg/x64dbg/blob/development/src/dbg/debugger.cpp#L2322-L2323](https://github.com/x64dbg/x64dbg/blob/development/src/dbg/debugger.cpp#L2322-L2323)
`char* dotInName = strrchr(basicName(), '.');
dotInName[0] = '\0';`
as you can see - no nullptr check before dereferencing dotInName.
to trigger the bug, a process has to be created where image name contains no dot, and `cmdLineExe`(`strstr(cmdline, pe32.szExeFile)`) has to evaluate to false(i.e. argv has to not contain image path)
fix:
add a nullptr check? I bytepatched the assembly to include a nullptr check(too lazy to compile from source), everything seems fine, but ofc you should do some more thorough checking

### Steps to reproduce

to reproduce, you have to
1) start a process where image name doesn't have a dot
2) supply(to the process in step 1) argv[0] that does not contain image name
easy reproduction: copy cmd.exe(or whatever you want really) somewhere but name it cmd(without extension), and then start it with some invalid argv[0](forfiles in some windows versions allows you to set argv[0])
paste in cmd:
`type C:\windows\system32\cmd.exe > C:\Users\user\Desktop\test\shit\cmd && start forfiles /p c:\windows\system32 /m notepad.exe /c "C:\Users\user\Desktop\test\shit\cmd 1"`
alternatively you can use kernel32.CreateProcessW and pass some shit in lpCommandLine
and then open Attach window in x64dbg
basically you want something like this:
[https://i.imgur.com/Ido0NY3.png](https://i.imgur.com/Ido0NY3.png)

**Comment by mrexodia** [comment]
Really appreciate the investigation! Should be fixed now. The code around there is quite a thing 😓 I also fixed two other potential issues, although I don't think those would manifest under real conditions.
